jsonobject/base.pyx

```diff
@@ -275,21 +275,17 @@ class JsonObjectBase(object):
 
     def __unwrap(self, key, value):
         property_ = self.__get_property(key)
-        try:
-            property_.validate(
-                value,
-                required=not self._validate_required_lazily,
-                recursive=False,
-            )
-        except TypeError:
```

Comment:
When could `TypeError` be thrown before? Is it not possible for it to happen now?

Reply:
I also didn't know, but it turns out that this was here to retain backwards compatibility for custom properties (defined outside this codebase) that don't include `recursive` in their signature. My latest commit is intended to fix this.